src/js/users/components/CreateUser.tsx

```diff
@@ -0,0 +1,56 @@
+import { DialogPortal, DialogTrigger } from "@radix-ui/react-dialog";
+import React, { useState } from "react";
+import { useMutation } from "react-query";
+import { Dialog, DialogContent, DialogOverlay, DialogTitle, Icon } from "../../base";
+import { StyledButton } from "../../base/styled/StyledButton";
+import { create } from "../api";
+import { CreateUserForm } from "./CreateUserForm";
+
+type NewUserProps = {
+    /** The user's handle or username */
+    handle: string;
+    /** The user's password */
+    password: string;
+    /** Whether the user will be forced to reset their password on next login */
+    forceReset: boolean;
+};
+
+/**
+ * A dialog for creating a new user
+ */
+export default function CreateUser() {
+    const [open, setOpen] = useState(false);
+    const [error, setError] = useState("");
```

Comment:
I think a lot of the error handling logic here is made redundant by the error handling features of `useMutation`. Take a look at the documentation of `useMutation` and see if you can trim down the error handling logic